drivers/sensor/bmm150/bmm150_trigger.c

```diff
@@ -173,3 +174,15 @@ int bmm150_trigger_mode_init(const struct device *dev)
 
 	return 0;
 }
+
+int bmm150_trigger_mode_uninit(const struct device *dev)
+{
+	struct bmm150_data *data = dev->data;
+	const struct bmm150_config *cfg = dev->config;
+
+#if defined(CONFIG_BMM150_TRIGGER_OWN_THREAD)
+	k_thread_abort(bmm150_thread_id);
+#endif
+
```

Comment:
Couldn't you just suspend the thread instead of aborting it ? I had mentioned in other comment, but you probably need to split this trigger initialization, one to just initialize data structures and other to handle pm logic. 

Then you *probably* can only suspend/resume the thread and semaphore